alacritty/src/display.rs

```diff
@@ -286,27 +287,27 @@ impl Display {
         size_info.cell_height = cell_height;
     }
 
-    /// Process resize events
-    pub fn handle_resize<T>(
+    /// Process update events
+    pub fn handle_update<T>(
         &mut self,
         terminal: &mut Term<T>,
         pty_resize_handle: &mut dyn OnResize,
         message_buffer: &MessageBuffer,
         config: &Config,
-        resize_pending: Resize,
+        update_pending: DisplayUpdate,
     ) {
         // Update font size and cell dimensions
-        if let Some(resize) = resize_pending.font_size {
-            self.font_size = match resize {
-                FontResize::Delta(delta) => max(self.font_size + delta, FONT_SIZE_STEP.into()),
-                FontResize::Reset => config.font.size,
+        if let Some(update) = update_pending.font {
+            self.font.size = match update {
+                FontUpdate::Delta(delta) => max(self.font.size + delta, FONT_SIZE_STEP.into()),
+                FontUpdate::Reset => config.font.size,
```

Comment:
We should pass the font explicitly instead of reading from a seemingly unrelated struct. Doing it like this makes it really hard to reason about where the font update is coming from and where it is sent to the display.

Reply:
Are you saying that we should do something like `FontUpdate::ResetFont(font) => font.size`?

Reply:
We'd probably need more than just font size if we don't want to store it on the Display, but yeah.